packages/geofire/src/GeoQuery.ts

```diff
@@ -277,20 +280,20 @@ export class GeoQuery {
    * @param query The geohash query.
    * @param queryState An object storing the current state of the query.
    */
-  private _cancelGeohashQuery(query: string[], queryState: QueryState): void {
-    const queryRef = this._firebaseRef.orderByChild('g').startAt(query[0]).endAt(query[1]);
```

Comment:
There seems to be a breaking change in the v9 SDK here, where it doesn't cancel the listeners.
My guess is that it's due to this being a new `queryRef` object.

Reply:
Hmm... not the SDK, but the implementation: the `on` function doesn't return the handler anymore, but returns an unsubscribe function (like Firestore). So instead of passing it to `off()`, we need to simply invoke it. Testing a fix now...

Reply:
Did you find a fix for this? It's the only thing blocking me right now.

Reply:
I think I figured it out. You were right, invoking it works, instead of passing it to `off()`: https://github.com/thatfiredev/geofire-js/commit/96c139042c45d5fa94cd809e9c0500755e0b6fc7